pkg/detectors/privatekey/privatekey.go

```diff
@@ -66,7 +66,7 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 
 		s1 := detectors.Result{
 			DetectorType: detectorspb.DetectorType_PrivateKey,
-			Raw:          []byte(token),
+			Raw:          []byte(match),
```

Comment:
We don't recommend updating the `Raw` field value, as it's used as an identifier and changing it can interfere with existing findings. Your solution is correct, but instead, you can use:

```go
s1.SetPrimarySecretValue(match)
```

This will instruct the engine to use `match` to determine the line number.